### ISSUE TITLE
AOTInductor doesn't support mutable custom operators

### ISSUE THREAD

**Issue by chengmengli06** [pytorch/pytorch#134835]
### 🐛 Describe the bug

I am trying compile a torchrec model which use fbgemm IntNBitTableBatchedEmbeddingBagsCodegen, which in turn use bounds_check_indices kernel,  and part of the fx graph is:
```
    ...
    %getitem : [num_users=1] = call_function[target=operator.getitem](args = (%permute_2d_sparse_data, 0), kwargs = {})
    %getitem_1 : [num_users=1] = call_function[target=operator.getitem](args = (%permute_2d_sparse_data, 1), kwargs = {})
    %view_2 : [num_users=1] = call_function[target=torch.ops.aten.view.default](args = (%getitem, [-1]), kwargs = {})
    %asynchronous_complete_cumsum : [num_users=1] = call_function[target=torch.ops.fbgemm.asynchronous_complete_cumsum.default](args = (%view_2,), kwargs = {})
    %auto_functionalized : [num_users=3] = call_function[target=torch._higher_order_ops.auto_functionalize.auto_functionalized](args = (fbgemm.bounds_check_indices.default,), kwargs = {rows_per_table: %b__tensor_constant0, indices: %getitem_1, offsets: %asynchronous_complete_cumsum, bounds_check_mode: 1, warning: %b__tensor_constant1, weights: None, B_offsets: None, max_B: -1})
    %getitem_4 : [num_users=1] = call_function[target=operator.getitem](args = (%auto_functionalized, 1), kwargs = {})
    %getitem_5 : [num_users=1] = call_function[target=operator.getitem](args = (%auto_functionalized, 2), kwargs = {})
    %getitem_6 : [num_users=1] = call_function[target=operator.getitem](args = (%auto_functionalized, 3), kwargs = {})
    ...
```
the kernel is not supported because it is mutable, what is the general solution to wrap a custom op which is mutable in inductor?

### Versions

torch=2.4.0
torchrec=0.8.0+cu121
fbgemm_gpu=0.8.0+cu121
```[tasklist]
### Tasks
```

cc @ezyang @chauhang @penguinwu @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4 @zou3519 @bdhirsh @desertfire @chenyang78 @mcarilli @eellison @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @kadeng @muchulee8 @ColinPeppler @amjames

**Comment by zou3519** [comment]
What do you mean by "the kernel is not supported because it is mutable"? Do you have an error message?

**Comment by ezyang** [comment]
My psychic debugging powers say that the user is complaining cudagraphs is inhibited

**Comment by zou3519** [comment]
The only cudagraphs inhibition I can repro is "skipping cudagraphs due to mutated [graph] inputs"

**Comment by chengmengli06** [comment]
![image](https://github.com/user-attachments/assets/8f55a425-6022-4c17-92e9-aa221863d3d3)
@zou3519 really surprised by your question, are you a developer of inductor? I think the developers should be familiar with the corner cases.

**Comment by zou3519** [comment]
TL;DR: the problem is AOTInductor doesn't support mutable custom operators

**Comment by chengmengli06** [comment]
any progress?

**Comment by angelayi** [comment]
Sorry, we haven't made any progress on supporting this yet, but I can take a look in 2-3 weeks.

**Comment by chengmengli06** [comment]
any progress? anyway to workaround, we get stuck here @angelayi

**Comment by desertfire** [comment]
Have you tried nightly? I think the problem might have been fixed.

**Comment by yushangdi** [comment]
close old issue. Please feel free to re-open if the issue persists.
